content/articles/license-plate-detection-and-recognition-using-opencv-and-pytesseract/index.md

```diff
@@ -0,0 +1,232 @@
+        perimeter = cv2.arcLength(c, True)
+        approx = cv2.approxPolyDP(c, 0.018 * perimeter, True)
+        if len(approx) == 4: 
+                screenCnt = approx
+```
+- `for c in cnts:` : We are creating a for loop over the contours we did sort. This is to find the best contour of our expected number plate.
+- `perimeter = cv2.arcLength(c, True)`: Perimeter is also referred to as arclength. We are using the arclength function to find it.
+- `cv2.approxPolyDP(c, 0.018 * perimeter, True)`: *ApproxPolyDP* approximates the curve of polygon with precision.
+- `if len(approx) == 4:`: chooses the contours with four sides as this will probably be our number plate.
+
+### Cropping the rectangular part identified as license plate
+```python                
+                x,y,w,h = cv2.boundingRect(c) 
+                new_img=image[y:y+h,x:x+w]
+                cv2.imwrite('./'+str(i)+'.png',new_img)
+                i+=1
+                break
+```
+- `x,y,w,h = cv2.boundingRect(c) `: This finds the coordinates of the part identified as the license plate.
+- `cv2.imwrite('./'+str(i)+'.png',new_img`: Stores the new image of the cropped number plate.
+- `break`: breaks the for loop.
+### Drawing the selected contour on the original image
+```python
+cv2.drawContours(image, [screenCnt], -1, (0, 255, 0), 3)
+cv2.imshow("image with detected license plate", image)
+cv2.waitKey(0)
+```
+- `cv2.drawContours(image, [screenCnt], -1, (0, 255, 0), 3)`: This draws the contour selected to be the number plate on our original image.
+- `cv2.imshow("image with detected license plate", image)`: Diplaying the final image that has a contour drawn over the number plate.
+
+### Extracting text from the image of the cropped license plate
+```python
+Cropped_loc = './7.png'
+cv2.imshow("cropped", cv2.imread(Cropped_loc))
+plate = pytesseract.image_to_string(Cropped_loc, lang='eng')
+print("Number plate is:", plate)
+cv2.waitKey(0)
+cv2.destroyAllWindows()
+
+```
+- `Cropped_loc = './7.png'`: This is the file name of the cropped image of the license plate.
+- `cv2.imshow("cropped", cv2.imread(Cropped_loc))`: We are displaying the image of the cropped license plate part.
+- `plate = pytesseract.image_to_string(Cropped_loc, lang='eng')`: We are passing the image of the cropped part of the license plate. We are then calling on pytesseract to extract the text on the image.
+- `print("Number plate is:", plate)`: We are printing out the extracted text.
+- `cv2.destroyAllWindows()`: We are closing all the open windows.
+### Results
+When the program has run successfully, its output is as follows:
+
+![Results](/engineering-education/license-plate-detection-and-recognition-using-opencv-and-pytesseract/results.gif)
+
+### Conclusion
+You now have all the skills required to create a program that detects and recognizes license plates. Capture more images of vehicles and input them into the program then sit back and watch as the program does its magic.
+
+! Happy coding

```

Comment:
Send the exclamation mark to the end of the phrase. Also, consider always adding a few references/ further reading links for the reader to get more content on the topic.